### ISSUE TITLE
Postgres: Parse alias expressions containing column definitions

### ISSUE THREAD

**Issue by derickl** [sqlfluff/sqlfluff#2929]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Running `sqlfluff lint` triggers a parse error on the following query
```sql
select
	tbl.columnA
	, t.col2
from schema1.table1 as tbl
cross join lateral somefunc(tb.columnB) AS t (col1 text, col2 bool)
;
```

### Expected Behaviour

SQLFluff should lint the above without error as it is valid [Postgres](https://www.postgresql.org/docs/14/sql-select.html) syntax.

### Observed Behaviour

```shell
== [rest2.sql] FAIL                                                                                                                                                     
L:   5 | P:  45 |  PRS | Line 5, Position 45: Found unparsable section: '(col1
                       | text, col2 bool)'
```

### How to reproduce

Run `sqlfluff lint --dialect postgres -n rest2.sql` on the following SQL
```sql
select
	tbl.columnA
	, t.col2
from schema1.table1 as tbl
cross join lateral somefunc(tb.columnB) AS t (col1 text, col2 bool)
;
```

### Dialect

postgres

### Version

`sqlfluff, version 0.11.1`

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
